crates/store/re_server/src/store/dataset.rs

```diff
@@ -50,6 +51,7 @@ impl Dataset {
                 name,
                 details,
                 partitions: HashMap::default(),
+                indexes: DatasetChunkIndexes::new(id),
```

Comment:
I believe it's incorrect to have the indexes in the inner object. We don't want to bump the modified timestamp when we change (e.g. delete) indexes. They should live the outer object instead. Might not change much since they have interior mutability, but just cleaner.